test_elasticsearch/test_helpers.py

```diff
@@ -26,22 +26,21 @@ def mock_process_bulk_chunk(*args, **kwargs):
 
 
 class TestParallelBulk(TestCase):
-    @mock.patch('elasticsearch.helpers._process_bulk_chunk', side_effect=mock_process_bulk_chunk)
+    @mock.patch('elasticsearch.helpers.actions._process_bulk_chunk', side_effect=mock_process_bulk_chunk)
     def test_all_chunks_sent(self, _process_bulk_chunk):
         actions = ({'x': i} for i in range(100))
         list(helpers.parallel_bulk(Elasticsearch(), actions, chunk_size=2))
 
         self.assertEquals(50, _process_bulk_chunk.call_count)
 
     @mock.patch(
-        'elasticsearch.helpers._process_bulk_chunk',
+        'elasticsearch.helpers.actions._process_bulk_chunk',
         # make sure we spend some time in the thread
         side_effect=lambda *a: [(True, time.sleep(.001) or threading.current_thread().ident)]
     )
     def test_chunk_sent_from_different_threads(self, _process_bulk_chunk):
         actions = ({'x': i} for i in range(100))
-        results = list(helpers.parallel_bulk(Elasticsearch(), actions, thread_count=10, chunk_size=2))
-
+        results = list(helpers.actions.parallel_bulk(Elasticsearch(), actions, thread_count=10, chunk_size=2))
```

Comment:
Because we've included `parallel_bulk` as an imported item into `__init__.py` that means this line doesn't need to change, right?

Reply:
Right, good catch. Mock needs absolute imports so I made changes for mock, but I didn't need to change that

Reply:
>  Mock needs absolute imports so I made changes for mock, but I didn't need to change that

yeah, I understood why those lines changed. I was just thinking about the call on this one line. Thanks for fixing!

Reply:
hm that broke the `2.7` build

Reply:
```
10:29:43 ======================================================================
10:29:43 FAIL: test_all_chunks_sent (test_elasticsearch.test_helpers.TestParallelBulk)
10:29:43 ----------------------------------------------------------------------
10:29:43 Traceback (most recent call last):
10:29:43   File "/usr/local/lib/python2.7/site-packages/mock/mock.py", line 1305, in patched
10:29:43     return func(*args, **keywargs)
10:29:43   File "/code/elasticsearch-py/test_elasticsearch/test_helpers.py", line 34, in test_all_chunks_sent
10:29:43     self.assertEquals(50, _process_bulk_chunk.call_count)
10:29:43 AssertionError: 50 != 49
```
One too few calls were made in 2.7

Reply:
This is a known issue, when it comes to multiprocessing and mock objects in python. Mock is not threadsafe and as a result causes issues.

Reply:
Gotcha, thanks

Reply:
Good to know. Resolving b/c your retest fixed it.